wolfSSL/src/chapter02.md

```diff
@@ -394,301 +394,1177 @@ Removes code for OAEP padding.
+
+#### USE_ECDSA_KEYSZ_HASH_ALGO
+
+Will choose a hash algorithm that matches the ephemeral ECDHE key size or the next highest available. This workaround resolves issues with some peers that do not properly support scenarios such as a P-256 key hashed with SHA512.
+
+#### WOLFSSL_ALT_CERT_CHAINS
+
+Allows CA's to be presented by peer, but not part of a valid chain. Default wolfSSL behavior is to require validation of all presented peer certificates. This also allows loading intermediate CA's as trusted and ignoring no signer failures for CA's up the chain to root. The alternate certificate chain mode only requires that the peer certificate validate to a trusted CA.
+
+#### WOLFSSL_SYS_CA_CERTS
+
+Allows wolfSSL to use trusted system CA certificates for verification when [`wolfSSL_CTX_load_system_CA_certs()`](group__CertsKeys.html#function-wolfssl_ctx_load_system_ca_certs) is called, either by loading them into wolfSSL certificate manager, or by invoking system authentication APIs. See [`wolfSSL_CTX_load_system_CA_certs()`](group__CertsKeys.html#function-wolfssl_ctx_load_system_ca_certs) for more details. This preprocessor macro is automatically set by the `--enable-sys-ca-certs` configure option.
+
+#### WOLFSSL_APPLE_NATIVE_CERT_VERIFICATION
+
+Enables the use of Apple's native trust APIs when authenticating TLS peer certificates. Requires [WOLFSSL_SYS_CA_CERTS](#WOLFSSL_SYS_CA_CERTS) to be defined. This macro does not need to be set by the user if building with `configure` or `CMake` on iOS or other apple devices, but should be explicitly set on MacOS if you wish to use the native verification methods.
+
+#### WOLFSSL_CUSTOM_CURVES
+
+Allow non-standard curves. Includes the curve "a" variable in calculation. Additional curve types can be enabled using [`HAVE_ECC_SECPR2`](#have_ecc_secpr2), [`HAVE_ECC_SECPR3`](#have_ecc_secpr3), [`HAVE_ECC_BRAINPOOL`](#have_ecc_brainpool) and [`HAVE_ECC_KOBLITZ`](#have_ecc_koblitz).
+
+#### HAVE_COMP_KEY
+
+Enables ECC compressed key support.
+
+#### WOLFSSL_EXTRA_ALERTS
+
+Enables additional alerts to be sent during a TLS connection. This feature is also enabled automatically when [`--enable-opensslextra`](#--enable-opensslextra) is used.
+
+#### WOLFSSL_DEBUG_TLS
+
+Enables additional debugging print outs during a TLS connection
+
+#### HAVE_BLAKE2
+
+Enables Blake2s algorithm support
+
+#### HAVE_FALLBACK_SCSV
+
+Enables Signaling Cipher Suite Value(SCSV) support on the server side. This handles the cipher suite 0x56 0x00 sent from a client to signal that no downgrade of TLS version should be allowed.
+
+#### HAVE_AEAD
+
+Implements the use of AEAD and is required for TLS 1.3.
+
+#### HAVE_AES_CBC
+
+Used In the case when Static RSA is needed.
+
+#### HAVE_AES_DECRYPT
+
+Enables AES to decrypt symmetric cypher.
```

Comment:
"cipher", not "cypher".  This setting controls whether the decrypt _direction_ of the various AES modes is supported.